apprise/plugins/NotifyFCM/__init__.py

```diff
@@ -69,7 +69,7 @@
     # We're good to go
     NOTIFY_FCM_SUPPORT_ENABLED = True
 
-except ImportError:
+except ImportError:  # pragma: no cover
```

Comment:
Not sure if this is needed; I mean it doesn't hurt.  The `cryptography` package shouldn't get installed in the `bare` environment:
[See here](https://github.com/panodata/apprise/blob/27f37f2ab9ed90a0de09339e7de6c51fd0b67a2d/.travis.yml#L31)

This was the way i was able to hit the edge cases of missing libraries without the `# pragma: no cover` option.  But it's possible your recent changes to tox maybe invalidate the edge case handling in the `.travis.yml` file?  Not sure.

Thoughts?

Reply:
I did not bother to put this line here because of this admonition:
```
apprise/plugins/NotifyFCM/__init__.py     190      3     76      0    99%   72-77
```
Let me know if this was inappropriate, then I will remove it. Should we add another test case to cover this situation?

Reply:
> The cryptography package shouldn't get installed in the `bare` environment.

Ah, this gets covered by the `bare` environment. I see. Thanks. I think I might have been on a short mission to have test cases that emulate this scenario by injecting errors on the `import` level. That may lead to not needing the `bare` environment any more. Do you think it is a good idea? Otherwise, I will back out of this and remove the pragma again.

Reply:
@caronc said:
> Not sure if the `pragma: no cover` is needed. [This situation is already covered by the `bare` environment.]

@amotl said:
> I might have been on a short mission to add test cases that emulate the "bare" scenario by injecting errors on the import level. Should we add another such test case to cover this situation?

See e674c5c51ae.

Reply:
Look at you go! :gear: :axe: :rocket: 

I mean i tried mocking the modules like you're doing and the tests failed for me at one point.  But you've done such an impressive job with even mocking the modules themselves, maybe the problems i had won't surface for you.   I wasn't sure if the effort to mock a module was worth just doing  a bare runner that hits those edge cases.

I'm fine with all of your approaches honestly.  If anything, i feel bad for how much effort you've been putting into this and just having me chime in now and then.  I think your `pragma: no cover` entries are perfectly fine. I missed that they were showing up as not being covered even with the `bare` reference... So no worries here.  I don't think TravisCI has a windows VM we can test truly having them fail, so no worries here at all.

Reply:
I'd also add that i got some REALLY weird errors with the TravisCI Test runner reloading modules.. the order to which they are reloaded is huge, and in some cases i just couldn't get it right (the cascading effect was a nightmare).   I'm not sure if the errors in your test cases after this tests changes you made now are just cycling through the pains i went through (or if just that fluke that keeps happening you already opened another ticket on).

I gave the one CI runner that failed a kick again to see if it helps (especially since all of the others passed)

Reply:
> I gave the one CI runner that failed a kick again to see if it helps (especially since all of the others passed).

Tests pass, but coverage declines. I am humbly asking if you can take a look because I think it has been introduced by cb7f51d82.

Reply:
> I mean i tried mocking the modules like you're doing and the tests failed for me at one point.
> I'd also add that i got some REALLY weird errors with the TravisCI Test runner reloading modules.

I hear you. I've observed similar issues when fiddling around with those aspects, while familiarizing myself with the code base and the test suite. I believe we should get rid of the root cause of this issue in the long-term. The root cause is that there is code on the module-level of plugin modules.

My proposal to get rid of this is to inline the corresponding code into a generic `setup()` method on the plugin class itself. In that way, everything can be handled much better than before, both at runtime and at test time. Let me know what you think about this idea.

Reply:
Yeah, not really sure where the problem got introduced, but this will fix it: https://github.com/caronc/apprise/pull/693/commits/7a48bf584de6a1a2e6fc510d67c9450877ff6b9e

Also some PEP8 issues got placed after my sloppy commit (fixed here: https://github.com/caronc/apprise/pull/693/commits/6d59cc2201aac3b6f194b8d26fa81035bc967819) 

Will roll these in with a PR i was working on #693